src/ape/types/__init__.py

```diff
@@ -284,7 +284,7 @@ class ContractLogContainer(list):
     def filter(self, event: "ContractEvent", **kwargs) -> List[ContractLog]:
         found_events = []
         for log in self:
-            if log.event_name == event.name:
+            if log.event_name == event.name and log.contract_address == event.contract.address:
```

Comment:
seems good, let me know if need help writing a test

Reply:
Yeah, would be appreciated!